File: leetcode/0387-first-unique-character-in-a-string/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 47 ms (78.17%)
     * MEMORY: 10.8 MB (46.38%)
    */
    int firstUniqChar(string s) {
        unordered_map<char, int> hashmap;
        for (char each : s) {
            ++hashmap[each];
        }
        for (int i = 0; i < s.size(); i++) {
            if (hashmap[s[i]] == 1) {
                return i;
            }
        }
        return -1;
    }
};

File: leetcode/0387-first-unique-character-in-a-string/solution.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        """
        RUNTIME: 128 ms (80%)
        MEMORY: 14.2 MB (18.48%)
        """
        hashmap = {}
        for i in s:
            if i not in hashmap:
                hashmap[i] = 1
            else:
                hashmap[i] += 1
        for i in s:
            if hashmap[i] == 1:
                return s.find(i)
        return -1
